test/readme.test.ts

```diff
@@ -0,0 +1,40 @@
+import { readFileSync } from 'node:fs';
+import path from 'node:path';
+
+import { z } from 'zod';
+
+import { loadConfig } from '../src/config';
+
+/*
+ * Recursively extracts keys from an object.
+ */
+const extractKeys = (item: any): string[] =>
+  typeof item === typeof {}
+    ? [...Object.keys(item), ...Object.values(item).flatMap((element) => extractKeys(element))]
+    : [];
+
+describe('checks README', () => {
+  it('checks that the README contains all configuration keys in airseeker.example.json', () => {
+    const config = loadConfig(path.join(__dirname, '../config'), 'airseeker.example.json');
+    expect(config).toBeDefined();
+
+    const readmeData = readFileSync(path.join(__dirname, '../README.md')).toString();
+    expect(readmeData).toBeDefined();
+
+    // console.log(config);
+    const keys = extractKeys(config).filter((item) => !z.coerce.number().safeParse(item).success);
+
+    const missingKeys = keys.filter((key) => readmeData.includes(`\n#### ${key}`));
+
```

Comment:
This does exact opposite. We want to filter all keys that are not in README data. The interpolation check like this will not work. I would imagine somtehing like this:

```ts
    const readmeHashtagTitles = readmeData
      .split('\n')
      .map((line) => /^#+ `(?<title>.+)`/.exec(line)?.groups?.title)
      .filter((title) => !!title);
    const readmeBacktickTitles = readmeData
      .split('\n')
      .map((line) => /^`(?<title>[^`]+)`/.exec(line)?.groups?.title)
      .filter((title) => !!title);
    const readmeTitles = [...readmeHashtagTitles, ...readmeBacktickTitles];
```

to capture all names starting with hastags or the ones that start just with "\`" (such as "url").

and get missing ones like this:
```ts
    const missingKeys = configKeys.filter(
      (configKey) => !readmeTitles.some((readmeTitle) => readmeTitle!.includes(configKey))
    );
```

but feel free to come up with a nicer heuristic.

Reply:
Sorry about that, I didn't test the inverse. I've added in the negation and changed the README a bit. If you prefer I can go with your approach above instead 👌